runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/provider/MuleExtensionModelDeclarer.java

```diff
@@ -164,7 +167,13 @@ class MuleExtensionModelDeclarer {
   private static final String TRACKING_NAMESPACE = "tracking";
   private static final String TRACKING_NAMESPACE_URI = "http://www.mulesoft.org/schema/mule/ee/tracking";
 
-  ExtensionDeclarer createExtensionModel() {
+  private final Optional<ComponentMetadataConfigurer.ComponentMetadataConfigurerFactory> configurerFactory;
+
+  public MuleExtensionModelDeclarer(ComponentMetadataConfigurer.ComponentMetadataConfigurerFactory configurerFactory) {
+    this.configurerFactory = ofNullable(configurerFactory);
+  }
+
```

Comment:
this shouldn't be optional. If null, it's a bug and it should fail. Also, using an Optional as an if(null) substitute is a documented anti-pattern. Optional is for communicating nullability, not for handling it internally